sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java

```diff
@@ -893,6 +906,19 @@ public void deleteDataset(String projectId, String datasetId)
           ignoreInsertIds);
     }
 
+    protected GoogleJsonError.ErrorInfo getErrorInfo(IOException e) {
+      GoogleJsonResponseException jsonCause = null;
```

Comment:
We can further simplify :)
protected GoogleJsonError.ErrorInfo getErrorInfo(IOException e) {
    if (!(e instanceof GoogleJsonResponseException)) {
        return null;
    }
    GoogleJsonError jsonError = ((GoogleJsonResponseException) e).getDetails();
    ...

Reply:
True, I have further simplified the method.